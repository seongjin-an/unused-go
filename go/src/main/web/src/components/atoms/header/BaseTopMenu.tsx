import React, { MouseEvent } from 'react';
import styled from 'styled-components';

interface IProps {}

export const BaseTopMenu: React.FC<IProps> = ({}) => {
    return (
        <StyledTopMenus>
            <li>나의 중고시새</li>
            <li>운동용품</li>
            <li>의류</li>
            <li>커뮤니티</li>
        </StyledTopMenus>
    );
};

const StyledTopMenus = styled.ul`
    margin: 0;
    padding: 0;
    display: flex;
    webkit-box-align: center;
    align-items: center;
    width: auto;
    height: 42px;
    cursor: pointer;
    list-style: none;

    & > li {
        font-family: PretendardBold;
        font-size: 18px;
        color: #f6f6f6;
    }

    //& > li:not(:last-child){
    //  margin-right: 105px;
    //}
    & > li:first-of-type {
        margin-right: 104px;
    }
    & > li:nth-of-type(2) {
        margin-right: 100px;
    }
    & > li:nth-of-type(3) {
        margin-right: 121px;
    }
`;
