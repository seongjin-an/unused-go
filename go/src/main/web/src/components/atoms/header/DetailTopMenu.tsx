import React, { MouseEvent } from 'react';
import styled, { css, keyframes } from 'styled-components';

interface IProps {
    isShow: boolean;
}

export const DetailTopMenu: React.FC<IProps> = ({ isShow }) => {
    return (
        <StyledDetailMenu isShow={isShow}>
            <ul>
                <li>
                    <ul>
                        <li>중고시세 한눈에보기</li>
                    </ul>
                </li>
                <li>
                    <ul>
                        <li>운동용품 전체</li>
                        <li>아령</li>
                        <li>바벨</li>
                        <li>원판/플레이트</li>
                        <li>벨트</li>
                        <li>악력기</li>
                        <li>밴드/짐볼</li>
                        <li>스트랩/보호대</li>
                    </ul>
                </li>
                <li>
                    <ul>
                        <li>의류 전체</li>
                        <li>재킷 &amp; 베스트</li>
                        <li>탑 &amp; 티셔츠</li>
                        <li>팬츠 &amp; 타이츠</li>
                        <li>양말</li>
                        <li>셋업</li>
                    </ul>
                </li>
                <li>
                    <ul>
                        <li>최신글</li>
                    </ul>
                </li>
            </ul>
        </StyledDetailMenu>
    );
};
const fadeIn = keyframes`
  from {
    opacity: 0;
    //display: none;
  }
  to {
    opacity: 1;
    //display: flex;
  }
`;
const fadeOut = keyframes`
  from {
    opacity: 1;
    //display: flex;
  }
  to {
    opacity: 0;
    //display: none;
  }
`;
const StyledDetailMenu = styled.div<{ isShow: boolean }>`
    //transition: opacity .15s ease-in-out, -webkit-transform .3s ease-in-out;
    display: none;
    ${({ isShow }) =>
        isShow
            ? css`
                  display: flex;
                  opacity: 1;
              `
            : css`
                  display: none;
                  opacity: 0;
              `};
    animation-name: ${({ isShow }) => isShow && fadeIn};
    animation-duration: 0.3s;
    //animation-iteration-count: 0;

    width: 100%;
    height: 372px;
    background-color: #341cff;
    border-bottom: 1px solid #fff;
    padding: 42px 0 0 238px;
    position: absolute;
    border-top: 1px solid #fff;
    top: 70px;
    left: 0;

    & > ul {
        display: flex;
        width: auto;
        height: auto;
    }

    & ul {
        margin: 0;
        padding: 0;
        list-style: none;
    }

    & > ul > li > ul {
        display: flex;
        -webkit-box-orient: vertical;
        -webkit-box-direction: normal;
        flex-direction: column;
        margin-right: 60px;
    }

    & > ul > li > ul > li {
        font-family: PretendardSemiBold;
        font-size: 18px;
        color: #ffffff;
        margin-bottom: 18px;
        cursor: pointer;
        transition: transform 0.3s ease-in-out, -webkit-transform 0.3s ease-in-out;
        &:hover {
            transform: translate(3px, 0);
            -ms-transform: translate(3px, 0);
            -webkit-transform: translate(3px, 0);
        }
    }
`;
