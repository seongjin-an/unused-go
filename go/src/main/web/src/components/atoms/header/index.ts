export { BaseTopMenu } from './BaseTopMenu';
export { DetailTopMenu } from './DetailTopMenu';
export { UserName } from './func/UserName';
export { FuncButton } from './func/FuncButton';
