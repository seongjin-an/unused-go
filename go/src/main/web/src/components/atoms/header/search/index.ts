export { CloseSearchButton } from './CloseSearchButton';
export { RecentItem } from './RecentItem';
export { SearchButton } from './SearchButton';
export { SearchInput } from './SearchInput';
