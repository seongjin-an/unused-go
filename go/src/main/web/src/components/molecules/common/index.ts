export { MenuNavigator } from './MenuNavigator';
export { Modal } from './Modal';
export { Logout } from './Logout';