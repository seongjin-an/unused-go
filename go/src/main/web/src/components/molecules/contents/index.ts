export {MainDash} from "./main/MainDash";
export {MainProduct} from './main/MainProduct'