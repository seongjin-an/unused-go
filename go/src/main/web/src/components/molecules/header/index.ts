export {Logo} from './Logo'
export {TopMenus} from './TopMenus'