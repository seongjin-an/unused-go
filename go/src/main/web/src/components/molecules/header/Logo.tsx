import React, { useContext } from 'react';
import styled from 'styled-components';
import { useRecoilState } from 'recoil';
import logoIcon from '../../../static/image/dark/common/topBar/img/img_logo.png';
import { Modal } from '../common';
import { modalState } from '../../../stores/modal';
import ModalContext from '../../../contexts/modalContext';

export const Logo: React.FC = () => {
  const [isOpen, setIsOpen] = useRecoilState<boolean>(modalState);
  // const {action: {setIsOpen}} = useContext(ModalContext)
  const handleOpen = () => {
    setIsOpen(true);
  };
  const handleClose = () => {
    setIsOpen(false);
  };
  return (
    <StyledLogoWrapper>
      <StyledLogoIcon onClick={handleOpen} />
      {/* <Modal isOpen={isOpen} onClose={handleClose}> */}
      {/*    <ModalBody> */}
      {/*        <h2>Title</h2> */}
      {/*        <p>Description</p> */}
      {/*    </ModalBody> */}
      {/* </Modal> */}
    </StyledLogoWrapper>
  );
};
const StyledLogoWrapper = styled.div`
  flex: 1.4;
  display: flex;
  align-items: center;
  justify-content: center;
`;
const StyledLogoIcon = styled.div`
  width: 128px;
  height: 43px;
  background: url(${logoIcon}) no-repeat;
`;
const ModalBody = styled.div`
  border-radius: 8px;
  box-shadow: 0 1px 3px 0 rgba(0, 0, 0, 0.1);
  background: #fff;
  max-height: calc(100vh - 16px);
  overflow: hidden auto;
  position: relative;
  padding-block: 12px;
  padding-inline: 24px;
`;
