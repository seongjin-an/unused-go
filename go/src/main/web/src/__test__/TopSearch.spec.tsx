import { configure, mount } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import React from 'react';
import { TopSearch } from '../components/molecules/header';

describe('TopSearch test~', () => {
  beforeEach(() => {
    console.log('=====moking test=====');
    configure({ adapter: new Adapter() });
  });
  afterEach(() => {
    console.log('=====done!=====');
  });
  it('should match snapshot Logo', () => {
    const handleCloseButton = jest.fn();
    const component = mount(<TopSearch isSearchShow handleCloseSearch={handleCloseButton} />);
    expect(component).toMatchSnapshot();
  });
  it('should render recent items', () => {
    const handleCloseButton = jest.fn();
    const component = mount(<TopSearch isSearchShow handleCloseSearch={handleCloseButton} />);
    const recentItems = component.find('RecentItem');
    expect(recentItems.length).toBe(5);
    const searchInput = component.find('SearchInput');
    const searchButton = component.find('SearchButton');
    searchInput.simulate('change', { target: { value: 'ansj' } });
    searchButton.simulate('click');
  });
});
