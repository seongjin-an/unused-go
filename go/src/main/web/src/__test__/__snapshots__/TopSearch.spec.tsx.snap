// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopSearch test~ should match snapshot Logo 1`] = `
<TopSearch
  handleCloseSearch={[MockFunction]}
  isSearchShow={true}
>
  <styled.div
    isSearchShow={true}
  >
    <div
      className="sc-bqiRlB gSYkBh"
    >
      <CloseSearchButton
        handleCloseSearch={[MockFunction]}
      >
        <styled.button
          onClick={[MockFunction]}
        >
          <button
            className="sc-pVTFL dgyIXM"
            onClick={[MockFunction]}
          />
        </styled.button>
      </CloseSearchButton>
      <styled.div>
        <div
          className="sc-egiyK fdVDyz"
        >
          <SearchInput>
            <styled.input
              placeholder="검색"
              type="text"
            >
              <input
                className="sc-iqseJM sjBMX"
                placeholder="검색"
                type="text"
              />
            </styled.input>
          </SearchInput>
          <SearchButton>
            <styled.button>
              <button
                className="sc-kDTinF eYxxHh"
              />
            </styled.button>
          </SearchButton>
        </div>
      </styled.div>
      <styled.div>
        <div
          className="sc-crHmcD iAvvCT"
        >
          <RecentItem>
            <styled.div>
              <div
                className="sc-jrQzAO kpqClJ"
              >
                <div
                  className="img"
                />
                <div
                  className="name"
                >
                  아령
                </div>
              </div>
            </styled.div>
          </RecentItem>
          <RecentItem>
            <styled.div>
              <div
                className="sc-jrQzAO kpqClJ"
              >
                <div
                  className="img"
                />
                <div
                  className="name"
                >
                  아령
                </div>
              </div>
            </styled.div>
          </RecentItem>
          <RecentItem>
            <styled.div>
              <div
                className="sc-jrQzAO kpqClJ"
              >
                <div
                  className="img"
                />
                <div
                  className="name"
                >
                  아령
                </div>
              </div>
            </styled.div>
          </RecentItem>
          <RecentItem>
            <styled.div>
              <div
                className="sc-jrQzAO kpqClJ"
              >
                <div
                  className="img"
                />
                <div
                  className="name"
                >
                  아령
                </div>
              </div>
            </styled.div>
          </RecentItem>
          <RecentItem>
            <styled.div>
              <div
                className="sc-jrQzAO kpqClJ"
              >
                <div
                  className="img"
                />
                <div
                  className="name"
                >
                  아령
                </div>
              </div>
            </styled.div>
          </RecentItem>
        </div>
      </styled.div>
    </div>
  </styled.div>
</TopSearch>
`;

exports[`header test~ should match snapshot Logo 1`] = `
<TopSearch
  handleCloseSearch={[MockFunction]}
  isSearchShow={true}
>
  <styled.div
    isSearchShow={true}
  >
    <div
      className="sc-egiyK bpXcVW"
    >
      <CloseSearchButton
        handleCloseSearch={[MockFunction]}
      >
        <styled.button
          onClick={[MockFunction]}
        >
          <button
            className="sc-furwcr hLvxrx"
            onClick={[MockFunction]}
          />
        </styled.button>
      </CloseSearchButton>
      <styled.div>
        <div
          className="sc-crHmcD dwlxTp"
        >
          <SearchInput>
            <styled.input
              type="text"
            >
              <input
                className="sc-kDTinF TWBvG"
                type="text"
              />
            </styled.input>
          </SearchInput>
          <SearchButton>
            <styled.button>
              <button
                className="sc-jrQzAO gofgDO"
              />
            </styled.button>
          </SearchButton>
        </div>
      </styled.div>
      <styled.div>
        <div
          className="sc-iqseJM djymIp"
        >
          <RecentItem>
            <styled.div>
              <div
                className="sc-pVTFL fwENdU"
              >
                <div
                  className="img"
                />
                <div
                  className="name"
                >
                  아령
                </div>
              </div>
            </styled.div>
          </RecentItem>
          <RecentItem>
            <styled.div>
              <div
                className="sc-pVTFL fwENdU"
              >
                <div
                  className="img"
                />
                <div
                  className="name"
                >
                  아령
                </div>
              </div>
            </styled.div>
          </RecentItem>
          <RecentItem>
            <styled.div>
              <div
                className="sc-pVTFL fwENdU"
              >
                <div
                  className="img"
                />
                <div
                  className="name"
                >
                  아령
                </div>
              </div>
            </styled.div>
          </RecentItem>
          <RecentItem>
            <styled.div>
              <div
                className="sc-pVTFL fwENdU"
              >
                <div
                  className="img"
                />
                <div
                  className="name"
                >
                  아령
                </div>
              </div>
            </styled.div>
          </RecentItem>
          <RecentItem>
            <styled.div>
              <div
                className="sc-pVTFL fwENdU"
              >
                <div
                  className="img"
                />
                <div
                  className="name"
                >
                  아령
                </div>
              </div>
            </styled.div>
          </RecentItem>
        </div>
      </styled.div>
    </div>
  </styled.div>
</TopSearch>
`;
